src/odin/maneuversbuilder.cc

```diff
@@ -2971,5 +2977,132 @@ void ManeuversBuilder::SetTraversableOutboundIntersectingEdgeFlags(std::list<Man
   }
 }
 
+void ManeuversBuilder::UpdateManeuverPlacementForInternalIntersectionTurns(
+    std::list<Maneuver>& maneuvers) {
+
+  auto is_turn_maneuver = [](DirectionsLeg_Maneuver_Type maneuver_type) -> bool {
+    switch (maneuver_type) {
+      case DirectionsLeg_Maneuver_Type_kSlightRight:
+      case DirectionsLeg_Maneuver_Type_kRight:
+      case DirectionsLeg_Maneuver_Type_kSharpRight:
+      case DirectionsLeg_Maneuver_Type_kUturnRight:
+      case DirectionsLeg_Maneuver_Type_kUturnLeft:
+      case DirectionsLeg_Maneuver_Type_kSharpLeft:
+      case DirectionsLeg_Maneuver_Type_kLeft:
+      case DirectionsLeg_Maneuver_Type_kSlightLeft:
+      case DirectionsLeg_Maneuver_Type_kRampRight:
+      case DirectionsLeg_Maneuver_Type_kRampLeft:
+      case DirectionsLeg_Maneuver_Type_kStayRight:
+      case DirectionsLeg_Maneuver_Type_kStayLeft:
+        return true;
+      default:
+        return false;
+    }
+  };
+
+  auto is_relative_straight = [](uint32_t turn_degree) -> bool {
+    return ((turn_degree >= kRelativeStraightTurnDegreeLowerBound) ||
+            (turn_degree <= kRelativeStraightTurnDegreeUpperBound));
+  };
+
+  Maneuver* prev_maneuver = nullptr;
+  for (auto& maneuver : maneuvers) {
+    // Verify that the previous maneuver is set
+    if (prev_maneuver) {
```

Comment:
```suggestion
    if (!prev_maneuver) {
      prev_maneuver = &maneuver;
      continue;
    }
      
```

Reply:
this happen for each maneuver - would only want to update in one location